pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/inv/BitSlicedRangeIndexCreator.java

```diff
@@ -0,0 +1,168 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.segment.local.segment.creator.impl.inv;
+
+import java.io.File;
+import java.io.IOException;
+import org.apache.pinot.segment.spi.ColumnMetadata;
+import org.apache.pinot.segment.spi.index.creator.CombinedInvertedIndexCreator;
+import org.apache.pinot.spi.data.FieldSpec;
+import org.roaringbitmap.RangeBitmap;
+
+import static org.apache.pinot.segment.spi.V1Constants.Indexes.BITMAP_RANGE_INDEX_FILE_EXTENSION;
+import static org.apache.pinot.spi.data.FieldSpec.DataType.*;
+
+public class BitSlicedRangeIndexCreator implements CombinedInvertedIndexCreator {
+
+  public static final int RANGE_INDEX_VERSION = 2;
+
+  private final RangeBitmap.Appender _appender;
+  private final File _rangeIndexFile;
+  private final FieldSpec.DataType _valueType;
+  private final long _minValue;
+  private final double _scale;
+
+  public BitSlicedRangeIndexCreator(File indexDir, ColumnMetadata metadata) {
+    _appender = RangeBitmap.appender(maxValue(metadata));
+    _rangeIndexFile = new File(indexDir, metadata.getColumnName() + BITMAP_RANGE_INDEX_FILE_EXTENSION);
+    _valueType = metadata.getDataType();
+    _minValue = minValue(metadata);
+    _scale = scale(metadata);
+  }
+
+  @Override
+  public void add(int dictId) {
+    _appender.add(dictId - _minValue);
+  }
+
+  @Override
+  public void add(int[] dictIds, int length) {
```

Comment:
throw `UnsupportedOperationException` to prevent callers from calling this method